src/renderer/html_handlebars/helpers/toc.rs

```diff
@@ -132,6 +136,9 @@ impl HelperDef for RenderToc {
 
                 out.write(">")?;
                 true
+            } else if is_foldable {
+                out.write("<a class=\"toggle\">")?;
```

Comment:
In general it isn't a good idea to wrap something into link (<a>) if that isn't an actual link but only the interactive element, browsers have the tons of tricky behavior around the links and well as the search engins, it could be hard to predict all the unintended impact caused by the empty link. Simple <span> with the same class, or with reasonable modifications could be ok for this case.

Reply:
Well, this only mimics the behaviour of the same line just below. This should be changed separately, then.

Reply:
If it's not appropriate to use an `<a>` link for non-hyperlink elements, that should also apply to the toggle-able chevrons which are currently using `<a class="toggle">`. The chevron, *and* its adjacent text, could use a `<button>` if preferred (or at least an aria `role="button"`).

Reply:
I don't think we'd want any of the default styling that should come with `<button>`, so I'd personally prefer `role="button"`. (Though again, this would have to be changed separately, so I will only add it to this PR if upstream is updated first.)